etl/assets.py

```diff
@@ -0,0 +1,71 @@
+import json
+import os
+from pathlib import Path
+
+from dagster import asset, define_asset_job
+
+from etl.resources import (
+    OpenAiEmbeddingModelResource,
+    OpenAiGenerativeModelResource,
+    WikipediaReaderResource,
+)
+
+
+@asset
+def wikipedia_articles_from_storage(
+    wikipedia_reader_resource: WikipediaReaderResource,
+):
+    """Materialize an asset of Wikipedia articles."""
+
+    return tuple(wikipedia_reader_resource.read())
+
+
+@asset
+def wikipedia_articles_with_summaries(
+    wikipedia_articles_from_storage,
+    openai_generative_model_resource: OpenAiGenerativeModelResource,
+):
+    """Materialize an asset of Wikipedia articles with summaries."""
+
+    enriched_wikipedia_articles = [
+        openai_generative_model_resource.enrich_record(wikipedia_article)
+        for wikipedia_article in wikipedia_articles_from_storage
+    ]
+
+    output_directory = Path(__file__).parent.absolute() / "data" / "output"
```

Comment:
Put this type of variable in a `ConfigurableResource` instead of hard-coding it.

Reply:
Here's the pattern I use:

```
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path

from dagster import ConfigurableResource, EnvVar


class OutputConfig(ConfigurableResource):  # type: ignore
    @dataclass(frozen=True)
    class Parsed:
        directory_path: Path

    directory_path: str

    @classmethod
    def default(cls, *, directory_path_default: Path) -> OutputConfig:
        return OutputConfig(directory_path=str(directory_path_default))

    @classmethod
    def from_env_vars(cls, *, directory_path_default: Path) -> OutputConfig:
        return cls(
            directory_path=EnvVar("OUTPUT_DIRECTORY_PATH").get_value(str(directory_path_default)),  # type: ignore
        )

    def parse(self) -> Parsed:
        return OutputConfig.Parsed(directory_path=Path(self.directory_path))
```

Reply:
Thanks, this pattern was helpful. I am also using it for `DataFilesConfig`.